mozdownload/scraper.py

```diff
@@ -310,13 +310,22 @@ def _download():
 
                 if log_level <= logging.INFO and content_length:
                     pbar.finish()
+
+            except requests.exceptions.HTTPError as ex:
```

Comment:
Please move the whole block into the `Exception` section below. Note that `HTTPError` is based on `Exception` so just have a type check and handle a 404 via a if condition. That way we can reduce the code duplication.